app/controllers/registered_applications_controller.rb

```diff
@@ -12,6 +12,7 @@ def show
     @user = current_user
     @registered_applications = @user.registered_applications
     @registered_application  = @registered_applications.find(params[:id])
+    @events = @registered_application.events.group_by(&:name)
```

Comment:
@sharadalt clarifying: Make use of @events in `app/views/registered_applications/show.html.erb`